packages/nimble-components/src/mapping/icon/tests/mapping-icon.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { html } from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { MappingIcon, mappingIconTag } from '..';
+import { fixture, type Fixture } from '../../../utilities/tests/fixture';
+import { waitForUpdatesAsync } from '../../../testing/async-helpers';
+import { Icon } from '../../../icon-base';
+
+describe('Icon Mapping', () => {
+    class TestIcon extends Icon {}
+
+    let element: MappingIcon;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+
+    // prettier-ignore
+    async function setup(): Promise<Fixture<MappingIcon>> {
+        return fixture<MappingIcon>(html`
+        <${mappingIconTag}
+            key="foo"
+            text="foo"
+            icon="nimble-test-icon">
+        </${mappingIconTag}>`);
+    }
+
+    it('should export its tag', () => {
+        expect(mappingIconTag).toBe('nimble-mapping-icon');
+    });
+
+    it('can construct an element instance', () => {
+        expect(document.createElement('nimble-mapping-icon')).toBeInstanceOf(
+            MappingIcon
+        );
+    });
+
+    it('resolves icon after it is defined', async () => {
+        ({ element, connect, disconnect } = await setup());
+        await connect();
+        await waitForUpdatesAsync();
+
+        expect(element.resolvedIcon).toBeUndefined();
+
+        DesignSystem.getOrCreate()
+            .withPrefix('nimble')
+            .register(
+                TestIcon.compose({
+                    baseName: 'test-icon',
+                    template: html`<nimble-xmark-icon></nimble-xmark-icon>`
```

Comment:
it'll be a mute point but fyi make sure to use the tag name constants and not hard code